nixos/modules/services/home-automation/home-assistant.nix

```
@@ -111,7 +111,9 @@ let
     hasAttrByPath (splitString "." component) cfg.config
     || useComponentPlatform component
     || useExplicitComponent component
-    || builtins.elem component (cfg.extraComponents ++ cfg.defaultIntegrations);
+    || builtins.elem component (
+      cfg.extraComponents ++ cfg.defaultIntegrations ++ map (x: x.domain) cfg.customComponents
```

Looks good to me, especially since https://developers.home-assistant.io/docs/creating_integration_manifest/#domain says

> This domain has to be unique and cannot be changed.

But
```suggestion
      cfg.extraComponents ++ cfg.defaultIntegrations ++ map (getAttr "domain") cfg.customComponents
```
is maybe nicer.